situp-plugins/common/src/test/java/com/amazon/situp/plugins/source/FileSourceTests.java

```diff
@@ -0,0 +1,142 @@
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.Queue;
+
+import static java.lang.String.format;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class FileSourceTests {
+    private static final String TEST_PIPELINE_NAME = "test-pipeline";
+    private static final int TEST_WRITE_TIMEOUT = 100;
+    private static final String TEST_FILE_PATH = "src/test/resources/test-file-source.tst";
+    private static final String FILE_DOES_NOT_EXIST = "file_does_not_exist";
+    private static final String FILE_CONTENT = "THIS IS A TEST";
+
+    @Test
+    public void testFileSourceWithEmptyFilePath() {
+        try {
+            new FileSource("", TEST_WRITE_TIMEOUT, TEST_PIPELINE_NAME);
+        } catch (RuntimeException ex) {
+            assertThat(ex.getMessage(), is(equalTo(format("Pipeline [%s] - path is a required attribute for file " +
+                    "source", TEST_PIPELINE_NAME))));
+        }
+    }
+
+    @Test
+    public void testFileSourceWithNullFilePath() {
+        try {
+            new FileSource(null, TEST_WRITE_TIMEOUT, TEST_PIPELINE_NAME);
+        } catch (RuntimeException ex) {
+            assertThat(ex.getMessage(), is(equalTo(format("Pipeline [%s] - path is a required attribute for file " +
+                    "source", TEST_PIPELINE_NAME))));
+        }
+    }
+
+    @Test
+    public void testFileSourceCreationWithValues() {
+        final FileSource fileSource = new FileSource(TEST_FILE_PATH, TEST_WRITE_TIMEOUT, TEST_PIPELINE_NAME);
+        assertThat(fileSource, notNullValue());
+    }
+
+    @Test
+    public void testFileSourceCreationWithNullPipelineName() {
+        try {
+            new FileSource(TEST_FILE_PATH, TEST_WRITE_TIMEOUT, null);
+        } catch (NullPointerException ex) {
+            assertThat(ex.getMessage(), is(equalTo("Pipeline name cannot be null")));
+        }
+    }
+
+    @Test
+    public void testFileSourceCreationUsingPluginSettings() {
+        final Map<String, Object> settingMap = ImmutableMap.of(
+                "path", TEST_FILE_PATH);
+        final PluginSetting pluginSetting = new PluginSetting("file", settingMap);
+        pluginSetting.setPipelineName(TEST_PIPELINE_NAME);
+        final FileSource fileSource = new FileSource(pluginSetting);
+        assertThat(fileSource, notNullValue());
+    }
+
+    @Test
+    public void testFileSourceCreationWithNullPluginSetting() {
+        try {
+            new FileSource(null);
+        } catch (NullPointerException ex) {
+            assertThat(ex.getMessage(), is(equalTo("PluginSetting cannot be null")));
+        }
+    }
+
+    @Test
+    public void testFileSourceWithNonExistentFile() {
+        final Queue<Record<String>> bufferQueue = new LinkedList<>();
+        final TestBuffer buffer = new TestBuffer(bufferQueue, 1);
```

Comment:
Could you not use another buffer we've already created in the source?

Reply:
I am assuming you are saying that we don't need the TestBuffer as we will not be using in this test

Reply:
Yes, no need for it since we already have a buffer written.